tests/end_to_end_test/restore_test.go

```diff
@@ -270,19 +276,17 @@ func TestSnapshotRestore(t *testing.T) {
 		t.Fatalf("unexpected stat() results on output.zip directory %v %v", st, err)
 	}
 
-	restoreFailDir := t.TempDir()
-
 	// Attempt to restore snapshot with an already-existing target directory
 	// It should fail because the directory is not empty
-	_ = os.MkdirAll(restoreFailDir, 0o700)
-
 	e.RunAndExpectFailure(t, "snapshot", "restore", "--no-overwrite-directories", snapID, restoreDir)
 
 	// Attempt to restore snapshot with an already-existing target directory
 	// It should fail because target files already exist
-	_ = os.MkdirAll(restoreFailDir, 0o700)
```

Comment:
@pavannd1 @julio-lopez - I couldn't figure out what these `os.MkdirAll` calls on an unused temp directory were supposed to do. I removed them and it didn't seem to affect the test. I figured it was probably obsolete or copy/paste code, but if you think there's another purpose I missed, I can add this back (with explanation if needed).

Reply:
You are probably right that it may be a remnant from copy paste or refactoring. Just skimming through it I cannot see the intent, but there may be something subtle. 
It makes sense to separate it from this PR. It seems unrelated.

Reply:
Well it is _somewhat_ related, in that the test case I'm adding here (`--no-overwrite-symlinks`) is juxtaposed with the other `--no-overwrite` testcases. Those other test cases are following this seemingly unnecessary pattern of 
```